src/ch01/array_copy_mpi.f90

```diff
@@ -28,11 +29,11 @@ program array_copy_mpi
 
 call mpi_barrier(mpi_comm_world, ierr)
 
-if (nproc == 0) then
-  call mpi_isend(array, size(array), mpi_int, 1, 1,&
+if (nproc == sender) then
+  call mpi_isend(array, size(array), mpi_int, receiver, 1,&
                  mpi_comm_world, request, ierr)
-else if (nproc == 1) then
-  call mpi_irecv(array, size(array), mpi_int, 0, 1,&
+else if (nproc == receiver) then
+  call mpi_irecv(array, size(array), mpi_int, sender, 1,&
```

Comment:
I never remember the argument names or orders for MPI functions, even ones I use all the time. It might be nice to create a named constant for the remaining `1` being passed to `isend` and `irecv` to help people like myself understand what info you're passing into the MPI API even if you don't explain it beyond that.